grakn-graql-shell/src/main/java/ai/grakn/graql/JsonSession.java

```diff
@@ -52,7 +52,11 @@
 
     private final Session session;
 
-    private final ExecutorService executor = Executors.newSingleThreadExecutor();
+    private final ExecutorService executor = Executors.newSingleThreadExecutor(runnable -> {
+        Thread thread = Executors.defaultThreadFactory().newThread(runnable);
+        thread.setDaemon(true);
+        return thread;
+    });
```

Comment:
@fppt I'd like to add a test for this, but I don't think there's a good way to do it. This executor was keeping a thread alive, which stopped the JVM exiting (causing the "10 minutes with no input" bug on Travis).